consensus/src/consensus/mod.rs

```diff
@@ -504,6 +509,52 @@ impl ConsensusApi for Consensus {
         self.headers_store.get_timestamp(self.get_sink()).unwrap()
     }
 
+    fn get_current_block_color(&self, hash: Hash) -> Option<bool> {
+        let mut queue: BinaryHeap<Reverse<SortableBlock>> = BinaryHeap::new();
+        let mut visited = BlockHashSet::new();
+
+        let initial_childrens = match self.get_block_children(hash) {
+            Some(children) => children,
+            None => return None,
+        };
+
+        for child in initial_childrens {
+            let block = self.get_block(child).unwrap();
+
+            if visited.insert(child) {
+                queue.push(Reverse(SortableBlock::new(child, block.header.blue_work)));
+            }
+        }
+
+        while let Some(Reverse(current_block)) = queue.pop() {
+            let current_block_hash = current_block.hash;
+            let current_block_data = self.get_ghostdag_data(current_block_hash).unwrap();
+
+            if self.services.reachability_service.is_chain_ancestor_of(current_block_hash, self.get_sink()) {
```

Comment:
Keep sink in a variable out of the loop. 
a. It is more efficient 
b. For consistency. Sink might change and might render a block you already visited to be the new merging chain block.